build/yarn-scripts.ts

```diff
@@ -67,19 +83,51 @@ function scalaclean() {
 function watch() {
   concurrently(
     [
-      "yarn scalawatch",
-      "webpack --watch --devtool nosources-source-map --config ./build/extension.webpack.config.js",
-      "yarn watch:svelte"
+      'yarn scalawatch',
+      'webpack --watch --devtool nosources-source-map --config ./webpack/ext-dev.webpack.config.js',
+      'yarn watch:svelte',
     ],
     {
       killOthers: ['failure', 'success'],
     }
   )
 }
 
+function package() {
+  const pkg_dir = 'dist/package'
+
+  // create .vscodeignore to not package all node_modules into the vsix
+  fs.writeFileSync(
+    path.join(pkg_dir, '.vscodeignore'),
+    `# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+**/node_modules/**/*
+!node_modules/@omega-edit/server/bin
+!node_modules/@omega-edit/server/lib
+`
+  )
+
+  execSync('yarn install', { cwd: pkg_dir })
+  execSync('yarn vsce package --out ../../', { cwd: pkg_dir })
```

Comment:
`yarn` running `yarn-scripts` that then run `yarn` is headache inducing.  I'm not a big fan of this, even if it means we have a longer `yarn scripts` `package` command.  Having `yarn` call these functions that then in turn call `yarn` again doesn't feel right to me.  I'd prefer that we don't do this and instead have a more complex `package` command.  It would be easier to track what's going on there.

Reply:
I mean this would depend on what others think, if others agree we should revert it back we can as this was approved before this PR. This would need to be a separate issue to resolve and would not be a part from this PR. @mbeckerle @arosien @stevedlawrence @nlewis05 @michael-hoke @stricklandrbls @rthomas320 what do you guys think?

Reply:
I'm not suggesting we toss the whole thing out, just avoid calling `yarn` in this supplemental scripts file.

Reply:
That makes sense but I also think having it all together makes easier to follow what is going on. If split up between running some stuff in the file and other in the scripts section it might get confusing.